engine/src/main/java/org/hibernate/search/query/engine/impl/HSQueryImpl.java

```diff
@@ -670,7 +678,32 @@ else if ( this.sort != null && projection != null ) {
 		);
 	}
 
+	private void validateSortFields(ExtendedSearchIntegrator extendedIntegrator) {
+		SortField[] sortFields = sort.getSort();
+		for ( SortField sortField : sortFields ) {
+			if ( sortField instanceof DistanceSortField ) {
+				validateDistanceSortField( extendedIntegrator, sortField );
+			}
+		}
+	}
 
+	private void validateDistanceSortField(ExtendedSearchIntegrator extendedIntegrator, SortField sortField) {
+		String field = sortField.getField();
+		boolean indexedField = false;
+		for ( Class<?> clazz : classesAndSubclasses ) {
+			IndexedTypeDescriptor indexedTypeDescriptor = extendedIntegrator.getIndexedTypeDescriptor( clazz );
+			FieldDescriptor fieldDescriptor = indexedTypeDescriptor.getIndexedField( field );
+			if ( fieldDescriptor != null ) {
+				indexedField = true;
+				if ( fieldDescriptor.getType() != Type.SPATIAL ) {
+					throw log.distanceSortRequiresSpatialField( field );
+				}
+			}
+		}
+		if ( !indexedField ) {
```

Comment:
The method name `validateDistanceSortField()` is a bit too specific with this check here.